Svg.Editor.Core/Tools/ColorTool.cs

```diff
@@ -240,14 +262,25 @@ public override async void Execute(object parameter)
 					return;
 				}
 
+				if(selectedColorIndex == 7 || selectedColorIndex == 4)
```

Comment:
That was my solution for the default text color problem that I mentioned you yesterday. When the user double-clicks a pin to enter some text, it'll use the actual value of the text-color-tool (`SelectedTextColorIndex`) to colorize that text. As we discussed, it should be white by default except in case of a yellow or white pin, the text should be black in those cases.
The default value of it is white and I check the selected color every time when the user selects a new color (not text color) and if the selected color is yellow(4) or white(7), it'll change the `SelectedTextColorIndex` to black.
It would be a little better if I would change it back when the selected color is not yellow or white:
```
if (selectedColorIndex == 7 || selectedColorIndex == 4)
{
	t.SelectedTextColorIndex = 0;
}
else
{
	t.SelectedTextColorIndex = 7;
}
```
Of course it can happen that the user creates a black pin, then a yellow one (so the text color is set to black) and she/he starts to add text to the black one.
But I cannot set the color depending on the color of the selected pin. If I would do so it would prevent the user from creating a text in the same color as the pin's color but we would also lose the opportunity to create a lot of eg. blue text  one after the other because it would create the text always in white or black first and then the user would have to change them to blue one by one.

And I used integers because the selected color and selected text color are stored as an int so it's the fastest approach I guess. Or would it be a better solution like this ⬇️ ?  It's definitely easier to read but it's slower.
```
var whiteIndex = Array.IndexOf(t.SelectableColorNames, "White");
var yellowIndex = Array.IndexOf(t.SelectableColorNames, "Yellow");
var blackIndex = Array.IndexOf(t.SelectableColorNames, "Black");
if (selectedColorIndex == whiteIndex || selectedColorIndex == yellowIndex)
{
	t.SelectedTextColorIndex = blackIndex;
}
else
{
	t.SelectedTextColorIndex = whiteIndex;
}
```

Reply:
what I do not like about this implementation is
1) it uses hardcoded indexes that won't work as soon as you configure other colors
2) the color tool should know nothing about pins - this is clearly a pin-related logic!
 
try to (again) reverse the dependency. 
Instead of the color tool deciding what to do for pins, let the color tool ask the pin tool which color to use (using some interface for example)

We may have to work through this together in a call tomorrow.
My brain is just too overwhelmed by all the stuff that was going on today. :-)

Reply:
Okay. Thank you!